Disallow non-file characters in pipeline names

parkertimmins opened the pull request:
Pipeline names may currently contain several non-alpha characters, such as comma and asterisk. Pipelines with such names cannot be fetched by id from the `GET _ingest/pipeline/` api as the comma is interpreted to mean multiple id queries, and the asterisk is treated as a star query. Thus these characters, along with other characters which are not allowed in index or alias names, should be disallowed from pipeline names.

closes https://github.com/elastic/elasticsearch/issues/104411

parkertimmins committed:
Disallow non-file characters in pipeline names

Pipeline names may currently contain several characters non-alpah
characters, such as comma and asterisk. Pipelines with such name cannot
be fetched by id from the `GET _ingest/pipeline/` api as the comma is
interpreted to mean multiple id queries, and the asterisk is treated as
a star query. Thus these characters, along with other characters which
are not allowed in index or alias names, should be disallowed from
pipeline names.

parkertimmins committed:
test fix

parkertimmins committed:
Update docs/changelog/108648.yaml

elasticsearchmachine commented:
Pinging @elastic/es-data-management (Team:Data Management)

elasticsearchmachine commented:
Hi @parkertimmins, I've created a changelog YAML for you.

parkertimmins committed:
Move pipeline name validation out of Pipeline.create

Pipeline.create can be called on existing pipeline. We want existing
pipelines with names that contains special characters to continue to
function. Thus only validation in put request before making pipeline.

parkertimmins committed:
Merge branch 'main' into validate-pipeline-name

parkertimmins committed:
remove unneeded import

parkertimmins commented:
\ci

parkertimmins committed:
Merge branch 'main' into validate-pipeline-name

nielsbauman commented:
Should we consult the Backwards Compatability Committee (BCC) on this?

cc @dakrone

dakrone commented:
This is definitely something we'd want to consider as breaking before merging. Also (separately), I wouldn't consider this large enough to introduce a brand new exception, we could likely use `IllegalArgumentException` for this. We can use `MetadataCreateIndexService.validateIndexOrAliasName` for the validation and construction of the `IllegalArgumentException` (we do this in a few places in the code).

parkertimmins commented:
I was thinking to discuss with the team whether or not this seemed breaking changy, but in that case I'll go ahead and open a BCC ticket.

parkertimmins committed:
Use IllegalArgumentException for bad pipeline name

Validating pipeline names is not worth creating a new exception, just use IllegalArgumentException .

lukewhiting commented:
Closing this for now as it's been superseeded by https://github.com/elastic/elasticsearch/pull/114837
